I ran the tests on gcc119 with something like `GOMAXPROCS=8 go test -v -timeout=1h`.  You could try unsetting `tags` in `test.sh` for aix and running it too.

As for jemalloc, @ayappanec you may know better. I tried building the tip of jemalloc on gcc119, it failed with a nonsensical: 
```
configure: error: Unsupported pointer size: 0
```

--- a/.github/workflows/ci-badger-tests.yml
+++ b/.github/workflows/ci-badger-tests.yml
@@ -15,6 +15,37 @@ permissions:
   contents: read
 
 jobs:
+  cross-compile:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - goos: linux
+            goarch: amd64
+          - goos: linux
+            goarch: arm64
+          - goos: darwin
+            goarch: amd64
+          - goos: darwin
+            goarch: arm64
+          - goos: windows
+            goarch: amd64
+          - goos: aix
+            goarch: ppc64
+          - goos: plan9
+            goarch: amd64
+    steps:
+      - uses: actions/checkout@v5
+      - name: Setup Go
+        uses: actions/setup-go@v6
+        with:
+          go-version-file: go.mod
+      - name: Cross-compile for ${{ matrix.goos }}/${{ matrix.goarch }}
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+        run: go build ./...
+
   badger-tests:
     runs-on: ubuntu-latest
     steps:

--- a/README.md
+++ b/README.md
@@ -254,6 +254,23 @@ Below is a list of known projects that use Badger:
 
 If you are using Badger in a project please send a pull request to add it to the list.
 
+### Platform Compatibility
+
+Badger uses OS-specific implementations for directory locking and `fsync` operations. On
+**POSIX-compliant systems** (Linux, macOS, BSD), these work as expected.
+
+For **non-POSIX platforms**, be aware of potential limitations:
+
+| Platform | File             | Notes                                                                |
+| -------- | ---------------- | -------------------------------------------------------------------- |
+| AIX      | `dir_aix.go`     | Directory `fsync` not supported; durability on crash may be affected |
+| Windows  | `dir_windows.go` | Uses different locking mechanism                                     |
+| Plan9    | `dir_plan9.go`   | No file locking support                                              |
+| WASM/JS  | `dir_other.go`   | No file locking support                                              |
+
+If you encounter issues on these platforms, review the corresponding `dir_*.go` source file for
+implementation details.
+
 ## Contributing
 
 If you're interested in contributing to Badger see [CONTRIBUTING](./CONTRIBUTING.md).

--- a/dir_aix.go
+++ b/dir_aix.go
@@ -138,7 +138,7 @@ func openDir(path string) (*os.File, error) { return os.Open(path) }
 // or see https://github.com/coreos/etcd/issues/6368 for an example.)
 func syncDir(dir string) error {
 	var err error
-	// TODO: AIX does not support fsync on a directory. Doing a full file system sync may be
-	// undesirable too (e.x os.Sync()).
+	// AIX does not support fsync on a directory.
+	// Data durability on crash may be affected.
 	return err
 }

